src/Chapter2.hs

```diff
@@ -858,7 +893,11 @@ and reverses it.
   function, but in this task, you need to implement it manually. No
   cheating!
 -}
-rewind = error "rewind: Not Implemented!"
+rewind :: [a] -> [a]
+rewind lst = case lst of
+  [] -> []
+  [x] -> [x]
+  xs -> last xs :  rewind (init xs)
```

Comment:
Your solution is correct! However, it is slow. In lists, it is quite slow to add anything at the end of the list. For instance, taking the last element of the list and removing it. That is why it is always better to rewrite it with the `:` cons. Remember the explanation with trains? 🚂 🚋 🚋 

That is why a more efficient solution is with the accumulator and the recursive function that will do the addition at the start of the list which is instant!

You can read a bit more about the `go` pattern here: https://kowainik.github.io/posts/haskell-mini-patterns#recursive-go

Reply:
At first I was a bit confused by the comment as I was indeed applying the cons operator - but I later understood the problem being (1) Getting `last xs` each time is going to be prohibitive because Haskell lists are Singly Linked lists (2) Doing the `init xs` each time will produce list copies.

Here is the revised `go` pattern solution:
```
rewind :: [a] -> [a]
rewind = go []
  where
    go :: [a] -> [a] -> [a]
    go xs (y:ys) = go (y:xs) ys
    go xs [] = xs
```